metaflow/datastore/task_datastore.py

```diff
@@ -165,14 +162,11 @@ def pathspec_index(self):
         idxstr = ','.join(map(str, (f.index for f in self['_foreach_stack'])))
         return '%s/%s[%s]' % (self._run_id, self._step_name, idxstr)
 
-    @property
-    def artifact_cache(self):
-        return self._artifact_cache
-
     @property
     def parent_datastore(self):
         return self._parent
 
+    @require_mode(None)
```

Comment:
why `@require_mode(None)`? Can't we just not have the decorator if the function is mode-agnostic?

Reply:
There was other stuff in that function to check access at some point. I think it is no longer there. We can always remove later if needed. I don't mind having some sort of gatekeeper on the functions to perform some checks though.